WordPress/src/main/java/org/wordpress/android/ui/DeepLinkingIntentReceiverViewModel.kt

```diff
@@ -29,117 +26,83 @@ class DeepLinkingIntentReceiverViewModel
     val toast = editorLinkHandler.toast
 
     /**
-     * Tracking URIs like `public-api.wordpress.com/mbar/...` come from emails and should be handled here
+     * This viewmodel handles the following URLs
+     * `wordpress.com/post/...`
+     * `wordpress.com/stats/...`
+     * `public-api.wordpress.com/mbar/`
+     * In these cases this function returns true
      */
-    fun shouldHandleTrackingUrl(uri: UriWrapper): Boolean {
-        // https://public-api.wordpress.com/mbar/
-        return uri.host == HOST_API_WORDPRESS_COM &&
-                uri.pathSegments.firstOrNull() == MOBILE_TRACKING_PATH
+    fun shouldHandleUrl(uri: UriWrapper): Boolean {
+        return isTrackingUrl(uri) ||
+                editorLinkHandler.isEditorUrl(uri) ||
+                statsLinkHandler.isStatsUrl(uri)
```

Comment:
I was thinking about this approach, I think you're right. It would be better. I haven't used it so far basically just because it's different than what we use in the deeplinking activity. I'll look into it.

Reply:
Implemented in d572755